src/services/database.py

```diff
@@ -14,20 +15,194 @@ def get_db_connection(db_file: str):
     con.row_factory = sqlite3.Row
     return con
 
-def initialize_users_db(erase=False) -> None:
+def initialize_users_db(reset: bool = False) -> None:
     '''
     Creates users database containing user credential information if it does not already exist.
 
-    If erase is set to true then the existing database is deleted a brand new one is created.
+    If reset is set to true then the existing database is deleted a brand new one is created.
 
     Keyword arguments:
-    erase -- flag indicating whether the database should be erased
+    reste -- flag indicating whether the database should be erased
     '''
     with get_db_connection(USERS_DB) as con:
-        if erase:
+        if reset:
             con.execute('DROP TABLE IF EXISTS users')
         con.execute('''CREATE TABLE IF NOT EXISTS users
                         (id INTEGER PRIMARY KEY AUTOINCREMENT,
                             username TEXT UNIQUE,
                             hashed_password TEXT)''')
         con.commit()
+
+def user_exists(username: str) -> bool:
+    """Returns true if user with input username exists in users database.
+
+    Args:
+        username (str): username to be checked
+
+    Returns:
+        bool: true if user exists in users database with input username otherwise false
+    """
+    with get_db_connection(USERS_DB) as con:
+        if con.execute('SELECT username FROM users WHERE username = ?', (username,)).fetchone():
+            return True
+        else:
+            return False
+
+def add_new_user(username: str, hashed_password: str) -> None:
```

Comment:
split up database.py into separate files for each type of data. That way, the "get_user_db" can be a common function that does not need to be passed any parameters. A place for common errors when repeating code, like in this file.